Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m

```diff
@@ -78,17 +75,53 @@ - (OCTCall *)callToChat:(OCTChat *)chat enableAudio:(BOOL)enableAudio enableVide
 
 - (BOOL)answerCall:(OCTCall *)call enableAudio:(BOOL)enableAudio enableVideo:(BOOL)enableVideo error:(NSError **)error
 {
-    return NO;
+    OCTToxAVAudioBitRate audioBitRate = (enableAudio) ? kDefaultAudioBitRate : kOCTToxAVAudioBitRateDisable;
+    OCTToxAVVideoBitRate videoBitRate = (enableVideo) ? kDefaultVideoBitRate : kOCTToxAVVideoBitRateDisable;
+
+    if (call.chat.friends.count == 1) {
+
+        OCTFriend *friend = call.chat.friends.firstObject;
+
+        BOOL status = [self.toxAV answerIncomingCallFromFriend:friend.friendNumber audioBitRate:audioBitRate videoBitRate:videoBitRate error:error];
+
+        [self.calls updateCallWithChat:call.chat updateBlock:^(OCTCall *call) {
+            call.status = (status) ? OCTCallStatusActive : OCTCallStatusInactive;
+        }];
+
+        if (status) {
+            self.audioEngine.friendNumber = friend.friendNumber;
+            [self.audioEngine startAudioFlow:error];
+        }
+
+        return status;
+    }
+    else {
+        // TO DO: Group Calls
+        return NO;
+    }
 }
 
 - (BOOL)togglePause:(BOOL)pause forCall:(OCTCall *)call error:(NSError **)error
 {
     OCTToxAVCallControl control = (pause) ? OCTToxAVCallControlPause : OCTToxAVCallControlResume;
-    call.status = (pause) ? OCTCallStatusPaused : OCTCallStatusActive;
 
     if (call.chat.friends.count == 1) {
+
         OCTFriend *friend = call.chat.friends.firstObject;
-        return [self.toxAV sendCallControl:control toFriendNumber:friend.friendNumber error:error];
+
+        BOOL status =  [self.toxAV sendCallControl:control toFriendNumber:friend.friendNumber error:error];
+
+        if (status) {
```

Comment:
Personally I prefer early return. I think it is much readable than
```
if {
    if {
        if {
...
```

It would be nice if you will use it here and in other places. In this case we won't have this stairs-like code.
Also you can use `pause` variable instead of comparing control.

```
- (BOOL)togglePause:(BOOL)pause forCall:(OCTCall *)call error:(NSError **)error
{
    OCTToxAVCallControl control = (pause) ? OCTToxAVCallControlPause : OCTToxAVCallControlResume;

    if (call.chat.friends.count == 1) {

        OCTFriend *friend = call.chat.friends.firstObject;

        if (! [self.toxAV sendCallControl:control toFriendNumber:friend.friendNumber error:error]) {
            return NO;
        }

        [self.calls updateCall:call updateBlock:^(OCTCall *callToUpdate) {
            callToUpdate.status = pause ? OCTCallStatusPaused : OCTCallStatusActive;
        }];
    }
    else {
        // TO DO: Group Calls
    }

    return YES;
}
```

This looks better for me. So, please, early returns whenever possible. :grinning: